erpnext/stock/doctype/material_request/material_request.js

```diff
@@ -339,50 +347,60 @@ frappe.ui.form.on("Material Request Item", {
 });
 
 erpnext.buying.MaterialRequestController = erpnext.buying.BuyingController.extend({
-	tc_name: function() {
+	tc_name: function () {
 		this.get_terms();
 	},
 
-	item_code: function() {
+	item_code: function (frm, cdt, cdn) {
 		// to override item code trigger from transaction.js
+		const row = locals[cdt][cdn];
+		if (row.item_code) {
+			frappe.db.get_value("Compliance Item", { "item_code": row.item_code }, "item_category")
+			    .then(item => {
+			    	if (item.message.item_category == "Dry Flower") {
+						frm.fields_dict.items.grid.set_column_disp("jar_type", true);
+						frm.fields_dict.items.grid.set_column_disp("unwind_direction", true);
+					}
+			    })
```

Comment:
Check your whitespace in these lines, there's mixed spaces and tabs. VSCode will let you change all your spaces to tabs.